src/pulling_gauge.rs

```diff
@@ -0,0 +1,75 @@
+use std::{collections::HashMap, fmt, sync::Arc};
+
+use crate::{
+    core::Collector,
+    proto::{Gauge, Metric, MetricFamily, MetricType},
+};
+use protobuf::RepeatedField;
+
+/// A prometheus gauge that reads the value from a provided function on every
+/// collect run. Used to expose metrics that are better to be polled as opposed
+/// to continuously set.
+#[derive(Clone)]
+pub struct PullingGauge {
+    desc: crate::core::Desc,
+    value: Arc<Box<dyn Fn() -> f64 + Send + Sync>>,
+}
+
+impl fmt::Debug for PullingGauge {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PullingGauge")
+            .field("desc", &self.desc)
+            .field("value", &"<opaque>")
+            .finish()
+    }
+}
+
+impl PullingGauge {
+    /// Create a new PullingGauge.
+    pub fn new<S1: Into<String>, S2: Into<String>>(
+        fq_name: S1,
```

Comment:
What does `fq` stands for here? Let's maybe expand this name to make it obvious?

Reply:
It was supposed to stand for "fully qualified". However, I've removed this prefix since the other types don't have it either. It's just `name` now.